src/Core/functions.php

```diff
@@ -26,6 +26,60 @@
     define('DS', DIRECTORY_SEPARATOR);
 }
 
+if (!function_exists('Cake\Core\pathCombine')) {
+    /**
+     * Combines parts with a forward-slash `/`.
+     *
+     * Skips adding a forward-slash if either `/` or `\` already exists.
+     *
+     * @param list<string> $parts
+     * @param bool|null $trailing Determines how trailing slashes are handled
+     *  - If true, ensures a trailing forward-slash is added if one doesn't exist
+     *  - If false, ensures any trailing slash is removed
+     *  - if null, ignores trailing slashes
+     * @return string
+     */
+    function pathCombine(array $parts, ?bool $trailing = null): string
+    {
+        $numParts = count($parts);
+        if ($numParts === 0) {
+            return '';
```

Comment:
Should this return `/` if `$trailing` is true?

Reply:
Yes, good catch. Adds some complexity to this case, but still think it makes the for loop easier to read.

Reply:
Thinking this through, this is not useful for creating the end-part of a path that you append to another part.

However, if you were generating a path prefix, this would be useful.